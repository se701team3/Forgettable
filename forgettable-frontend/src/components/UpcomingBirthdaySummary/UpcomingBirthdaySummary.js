import React from 'react';
import {Avatar} from '@mui/material';
import classes from './UpcomingBirthdaySummary.module.css';
import {getBirthdayDate} from '../../functions/dateFormatter';
import UnknownDetail from '../UnknownDetail/UnknownDetail';
import PropTypes from 'prop-types';
import {getImageSrcFromBuffer} from '../../functions/getImageSrcFromBuffer';

/*
 * Component for displaying information for upcoming birthdays.
 * Based off of EncounterCardSummary and PersonCardSummary components
 * orginally authored by Mercury Lin (lin8231)
 *
 * Author: Aaron Song (ason720)
 */
const UpcomingBirthdaySummary = (props) => {
  return (
    <div
      className={classes.UpcomingBirthdaySummary}
      onClick={props.onClick}
      data-testid="container-card"
    >
      <div className={classes.ContentContainer}>
        <div className={classes.HeaderContainer}>
          <Avatar
            alt={props.name}
            src={getImageSrcFromBuffer(props.img)}
            sx={{
              height: '70px',
              width: '70px',
              marginRight: '14px',
              backgroundColor:
                getComputedStyle(document.body).getPropertyValue('--prmry'),
              fontSize:
                getComputedStyle(document.body)
                    .getPropertyValue('--text-xxlarge'),
            }}
          />
          <div className={classes.IdentityInfoConatiner}>
            <h3 data-testid="name-element">{props.name}</h3>
          </div>
        </div>
        <div className={classes.DetailsContainer}>
          <p>
            {'Birthday: '}
            {props.birthday ?
              <span data-testid="birthdate-element">
                {getBirthdayDate(props.birthday)}
              </span> :
              <UnknownDetail />}
          </p>
        </div>
      </div>
    </div>
  );
};

UpcomingBirthdaySummary.propTypes = {
  id: PropTypes.string,
  name: PropTypes.string.isRequired,
  img: PropTypes.string,
  birthday: PropTypes.instanceOf(Date),
  onClick: PropTypes.func.isRequired,
};

export default UpcomingBirthdaySummary;
