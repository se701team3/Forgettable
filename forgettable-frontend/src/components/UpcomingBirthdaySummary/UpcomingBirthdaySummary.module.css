@import  '../../colours.css';

.UpcomingBirthdaySummary{
    height: 150px;
    width:240px;
    cursor: pointer;
    border-radius: var(--radius-card);
    background-color: var(--card);
    transition: 0.3s;
    display: flex;
    padding: 20px 22px;
    box-sizing: border-box;
}

.UpcomingBirthdaySummary:hover {
    background-color: var(--hilit);
}

.ContentContainer {
    width: 100%;
    height: 100%;
    display: flex;
    flex-direction: column;
}

.HeaderContainer {
    width: 100%;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
}

.IdentityInfoConatiner {
    flex: 1;
    display: flex;
    flex-direction: column;

}

.IdentityInfoConatiner > h3 {
    font-size: var(--text-medium);
    color: var(--txt1);
    font-weight: var(--font-semibold);
    margin: 0;
    text-align: left;
    overflow: hidden;
    text-overflow: ellipsis;
    display: -webkit-box;
    -webkit-line-clamp: 1; /* number of lines to show */
            line-clamp: 1; 
    -webkit-box-orient: vertical;
}

.IdentityInfoConatiner > p {
    font-size: var(--text-card-body-normal);
    margin: 0;
    color: var(--txt2);
    text-align: left;
    overflow: hidden;
    text-overflow: ellipsis;
    display: -webkit-box;
    -webkit-line-clamp: 2; /* number of lines to show */
            line-clamp: 2; 
    -webkit-box-orient: vertical;
}

.DetailsContainer {
    flex: 1;
    display: flex;
    flex-direction: column;
}

.DetailsContainer > p {
    color: var(--txt2);
    font-size: var(--text-xsmall);
}

.DescriptionText {
    margin-top: -2px;
    overflow: hidden;
    text-align: left;
    text-overflow: ellipsis;
    display: -webkit-box;
    -webkit-line-clamp: 3; /* number of lines to show */
            line-clamp: 3; 
    -webkit-box-orient: vertical;
}

.IdentityInfoConatiner > h3 {
    text-align: left;
    text-overflow: ellipsis;
    overflow: hidden;
    max-width: 110px;
}