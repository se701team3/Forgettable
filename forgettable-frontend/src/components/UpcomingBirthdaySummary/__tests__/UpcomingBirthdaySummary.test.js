import React from 'react';
import {render, fireEvent, cleanup} from '@testing-library/react';

import ShallowRenderer from 'react-test-renderer/shallow';

import UpcomingBirthdaySummary from '../UpcomingBirthdaySummary';
import moment from 'moment';

jest.mock('../../../services');

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));

it('renders UpcomingBirthdaySummary UI with correct hierarchy', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<UpcomingBirthdaySummary
    name='Bob'
    onClick={() => { }} />);
  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});

afterEach(cleanup);

it('successfully fires event when card is clicked', () => {
  const handleClick = jest.fn();

  const {getByTestId} = render(
      <UpcomingBirthdaySummary
        name='Bob'
        birthday={moment(1648698811606).toDate()}
        onClick={handleClick}
      />);

  const node = getByTestId('container-card');

  fireEvent.click(node);
  expect(handleClick).toHaveBeenCalledTimes(1);
});
