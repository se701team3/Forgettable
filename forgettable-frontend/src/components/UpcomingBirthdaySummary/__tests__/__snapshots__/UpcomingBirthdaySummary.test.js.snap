// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders UpcomingBirthdaySummary UI with correct hierarchy 1`] = `
<div
  className="UpcomingBirthdaySummary"
  data-testid="container-card"
  onClick={[Function]}
>
  <div
    className="ContentContainer"
  >
    <div
      className="HeaderContainer"
    >
      <ForwardRef(Avatar)
        alt="Bob"
        src={null}
        sx={
          Object {
            "backgroundColor": "",
            "fontSize": "",
            "height": "70px",
            "marginRight": "14px",
            "width": "70px",
          }
        }
      />
      <div
        className="IdentityInfoConatiner"
      >
        <h3
          data-testid="name-element"
        >
          Bob
        </h3>
      </div>
    </div>
    <div
      className="DetailsContainer"
    >
      <p>
        Birthday: 
        <UnknownDetail />
      </p>
    </div>
  </div>
</div>
`;
